src/values.rs

```diff
@@ -0,0 +1,68 @@
+//! Types representing for data which will be retrieved from the database.
+//! Currently this data is expected to look like a JSON object but this may be
+//! changed in the future. Driver authors must cast the data they retrieve from
+//! the database to these types.
+
+use std::collections::BTreeMap;
+use structure::Collection;
+
+/// Represents a JSON pointer to a document property.
+pub type Pointer = Vec<String>;
+
+/// Various value types. Based on types in the [JSON standard][1] (see section
+/// 5).
+///
+/// [1]: http://ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf
+pub enum Value {
+  /// The abscense of any value.
+  Null,
+  /// True or false.
+  Boolean(bool),
+  /// A numeric value, float, integer, whatever.
+  Number(f64),
+  /// A list of characters.
+  String(String),
+  /// A list of values.
+  Array(Vec<Value>),
+  /// A map of key/value pairs.
+  Object(BTreeMap<String, Value>),
```

Comment:
`BTreeMap` probably isn't performant enough for our case. Do `Vec<(String, Value)>`